static/js/list_rendering.js

```diff
@@ -0,0 +1,196 @@
+var list_render = (function () {
+    var DEFAULTS = {
+        LOAD_COUNT: 20,
+        instances: {},
+    };
+
+    // @params
+    // container: jQuery object to append to.
+    // list: The list of items to progressively append.
+    // opts: An object of random preferences.
+    var func = function ($container, list, opts) {
```

Comment:
Can we change `func` to something like `get_renderer`?  I also think we might want to separate two concerns here by having two functions:

memoization: make that be a function that simply keeps track of objects and has the 3/4 lines of code to make that happen
renderer: make that function be the object with the "guts" and don't have it track objects